localstack/utils/generic/file_utils.py

```diff
@@ -29,3 +30,15 @@ def parse_config_file(file_or_str: str, single_section: bool = True) -> Dict:
         result = result[sections[0]]
 
     return result
+
+
+def cache_dir() -> Path:
+    if is_windows():
+        return Path("%LOCALAPPDATA%", "cache", "localstack")
+    if is_mac_os():
+        return Path.home() / "Library" / "Caches" / "localstack"
+    if is_linux():
+        string_path = os.environ.get("XDG_CACHE_HOME")
+        if string_path and os.path.isabs(string_path):
+            return Path(string_path)
+        return Path.home() / ".cache" / "localstack"
```

Comment:
what's the default behavior? maybe we should just use line 44 as fallback? otherwise the output type would be `Optional[Path]`, or raise an error that the system is unknown?